### DISCUSSION TITLE
Associate macOS file types with the program installed into Wineskin

### DISCUSSION THREAD

**Discussion by Tosyk** [The-Wineskin-Project/WineskinServer#258] [General]
Hi, I have a viewer which I use in windows. It opens fbx files among others. I've made a wineskin app for it to use on macOS. How can I associate my fbx file types on my mac so I can open them with double click in that app?
When I open FBX files on my mac with just the app I've created I see this message:
<img width="368" alt="image" src="https://user-images.githubusercontent.com/24209966/167840844-9de687ec-f3e5-4c07-a654-91d2182b5ef2.png">

p.s.: I see this here but it is only for linux: https://wiki.winehq.org/FAQ#How_do_I_associate_a_native_program_with_a_file_type_in_Wine.3F

**Comment by emendelson** [comment]
My guess is that you need to do two things: First, use the method that you found for Linux. Then, cmd-click an .fbx file on your Mac; chose Get Info, then Open With, select your wineskin app and click "Change all" so that all .fbx files will open with your wineskin app. Let us know if it works.

(And I'm curious - which file viewer are you using?)

**Reply by Tosyk** [reply]
Thanks for the suggestion! The thing is I'm very new to the scripting on mac and I've never done it with linux. So these words:
> First, use the method that you found for Linux.

mean nothing to me. There's an info about creating .reg files but does macOS supports them at all?

p.s.: tool I'm using is Noesis

**Comment by emendelson** [comment]
Run the Wineskin app inside your wrapper; choose Tools->Registry Editor. Look in HKCR (HKEY_CLASSES_ROOT) for entries like .txt to see the kind of thing you need to add for .fbx files (probably at least two entries, one for .fbx and the other for something like fbxfile, which will need to include the "open" command that runs your Noesis program with the current file ("%1"). I don't remember the details, but it's not difficult at all. Just look at some examples, like .txt, .htm, etc.

**Reply by Tosyk** [reply]
<img width="833" alt="image" src="https://user-images.githubusercontent.com/24209966/167859929-c005a5e6-3c1c-4057-9a99-b83cb659c2fc.png">
I've created this at last. But it doesn't work anyway

**Comment by emendelson** [comment]
Did you also create an .fbx entry that contains "fbxfile" in the way that .txt contains something like txtfile or similar? Both are needed.

**Reply by Tosyk** [reply]
yes, I did everything. but still I see this
<img width="354" alt="image" src="https://user-images.githubusercontent.com/24209966/167872851-769c97a9-8d70-4f57-9313-ee37a5770533.png">

**Reply by Tosyk** [reply]
it actually has this ui for such process even, but it doesn't work still
<img width="755" alt="image" src="https://user-images.githubusercontent.com/24209966/167873282-81a62cff-9e47-4e61-bc87-f5c1f6d7df43.png">

**Reply by Tosyk** [reply]
I also found this, but don't know how to apply it to make Noesis a default fbx viewer in my mac
https://wiki.winehq.org/MacOS_FAQ#How_to_create_shortcut.2C_launcher.2C_or_.app_to_start_a_given_.exe.3F

**Reply by Tosyk** [reply]
I tried it the same way with Notepad ++ and it worked. I wonder why it doesn't with Noesis?

**Reply by emendelson** [reply]
Put your wineskin app and a sample file on a download site; post a link here; and I'll experiment with it over the weekend.

**Reply by Tosyk** [reply]
thank you. here's the link:
https://drive.google.com/drive/folders/1L6nQKFIVmSCXY_kelfqcexZ6FrFew-qc?usp=sharing
to view the file you can also open Noesis and drop fbx file into its preview window

**Reply by Tosyk** [reply]
I found the same behavior I have here:
https://www.reddit.com/r/wine_gaming/comments/fuawmx/file_associations_in_wine/
but I don't understand how I can use it to fix my issue, any help?

**Reply by Tosyk** [reply]
hi, any thoughts on the issue?

**Reply by emendelson** [reply]
I've been on the road and didn't get to this. Will try to look at it in the next day or two and report back.

**Comment by emendelson** [comment]
All you need to do is add `"%1"` (including the quotation marks) to the end of the Windows .EXE field in the advanced tab, so that the field looks like this:

`"C:\Program Files\noesis\Noesis.exe" "%1"`

Then just drop a file on the app. 

At least this works when dropping files from the desktop on to the app (also on the desktop). You may need to use the Config Utility tool to map other directories to drive letters, but that's easy to figure out.

**Reply by Tosyk** [reply]
it's great man! thank you.
the only issue I have is that I can't open this file from google desktop folder which is here:
```
/Volumes/GoogleDrive/My Drive/cgig/downloads/tools/gms/_engine/clausewitz/_multitools/io_ck3_character/io_ck3_character.fbx
```
but I can open same file from here:
```
/Users/usr/Desktop/io_ck3_character.fbx
```
or from here also works:
```
/Users/usr/LebrancheMullet/io_ck3_character.fbx
```
in my Desktop Google Drive I have this enabled:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/24209966/168688399-2ed046b2-fd18-4fe4-9443-def2ba76fffe.png">
I'm working usually with google drive to share my work with colleagues. I'm not sure if you can help me but if you do that would be awesome

**Comment by emendelson** [comment]
I don't use Google Drive so I have no idea of what the answer might be. Sorry.
